node/resolvers/sponsoredProducts.ts

```diff
@@ -66,6 +66,7 @@ export async function sponsoredProducts(
   const adResponse = await ctx.clients.adServer.getSponsoredProducts({
     count: SPONSORED_PRODUCTS_COUNT,
     searchParams: getSearchParams(args),
+    userId: args.anonymousId,
```

Comment:
Please, test what would happen if `anonymousId` is `undefined`. I'd like to see the AdServer's behavior in that case.

Reply:
Currently it is receiving it undefined, since we're not passing anything. There's a `#[serde(default)]` attribute on that field that initializes it as an empty string.

Reply:
Yea, but your change now sends `userId: undefined`, which is different from not sending a `userId`, right?

I want to see how the server behaves in this scenario.

Reply:
Good point! I sent a GraphQL request without `anonymousId` and Ad Server behaved correctly:

```js
{
  adRequestId: 'd455eb84-c39e-41b6-9928-86a79a99cf5d',
  adResponseId: 'e9481791-75cd-4f01-b3c8-7f21b9016f05',
  userId: '',
  abTestId: '64543ba10604511f33708224-null',
  sponsoredProducts: [
    {
      campaignId: '5adaa995-45d1-4fd3-934a-1506259b55b4',
      adId: 'f45789bf-c681-4bd9-9ff6-00c4fed12b94',
      productId: '1',
      actionCost: 0.5
    }
  ]
}
```

This is because `JSON.stringify` removes undefined entries from objects (there's no undefined in JSON).

Reply:
OK then, perfect!